crates/ruff_linter/src/rules/pylint/rules/useless_import_alias.rs

```diff
@@ -52,6 +53,73 @@ pub(crate) fn useless_import_alias(checker: &mut Checker, alias: &Alias) {
     if alias.name.as_str() != asname.as_str() {
         return;
     }
+    // While the fix is specified as always available,
+    // the presence of a user-specified required import (I002)
+    // with a useless alias causes an infinite loop. So
+    // in this case we keep the diagnostic but omit the fix.
+    // See https://github.com/astral-sh/ruff/issues/14283
+    let required_imports = &checker.settings.isort.required_imports;
+    if !required_imports.is_empty() {
+        let semantic_alias = semantic::Alias {
+            name: alias.name.as_str().to_owned(),
+            as_name: Some(asname.as_str().to_owned()),
+        };
+        if required_imports.contains(&semantic::NameImport::Import(semantic::ModuleNameImport {
+            name: semantic_alias,
+        })) {
+            let diagnostic = Diagnostic::new(UselessImportAlias, alias.range());
+            checker.diagnostics.push(diagnostic);
+            return;
+        }
+    }
+
+    let mut diagnostic = Diagnostic::new(UselessImportAlias, alias.range());
+    diagnostic.set_fix(Fix::unsafe_edit(Edit::range_replacement(
+        asname.to_string(),
+        alias.range(),
+    )));
+    checker.diagnostics.push(diagnostic);
+}
+
+/// PLC0414
+pub(crate) fn useless_importfrom_alias(
+    checker: &mut Checker,
+    alias: &Alias,
+    module: Option<&str>,
+    level: u32,
+) {
+    let Some(asname) = &alias.asname else {
+        return;
+    };
+    if alias.name.contains('.') {
```

Comment:
Why this part?

Reply:
I don't know... This seems to have been there since the original implementation: https://github.com/harupy/ruff/blob/d765fbe5ddca1fa7b735256ebb303799a2e7aa95/src/pylint/plugins.rs#L16

I don't think it's possible to have an alias with a dot in it... maybe it's supposed to be an early return? Is checking for a dot faster than comparing two strings? Certainly can't be that much of a speed up... Anyway, happy to remove it.

Reply:
Thanks. Do you mind removing it (here and in the linked location) in a separate PR?

Reply:
Would it make sense to implement helper methods on the `isort` settings to test if it is a required import to avoid duplicating the logic?

Reply:
Ok! Refactored to use new helper methods on `isort` settings, and I will follow up after this PR is merged to get rid of the check for "."